repl/account_commands.go

```diff
@@ -0,0 +1,129 @@
+package repl
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/spacemeshos/CLIWallet/log"
+	apitypes "github.com/spacemeshos/api/release/go/spacemesh/v1"
+	"github.com/spacemeshos/ed25519"
+	gosmtypes "github.com/spacemeshos/go-spacemesh/common/types"
+)
+
+func (r *repl) chooseAccount() {
+	accs := r.client.ListAccounts()
+	if len(accs) == 0 {
+		r.createAccount()
+		return
+	}
+
+	fmt.Println(printPrefix, "Choose an account to load:")
+	accName := multipleChoice(accs)
+	account, err := r.client.GetAccount(accName)
+	if err != nil {
+		panic("wtf")
+	}
+	fmt.Printf("%s Loaded account alias: `%s`, address: %s \n", printPrefix, account.Name, account.Address().String())
+
+	r.client.SetCurrentAccount(account)
+}
+
+func (r *repl) createAccount() {
+	fmt.Println(printPrefix, "Create a new account")
+	alias := inputNotBlank(createAccountMsg)
+
+	ac := r.client.CreateAccount(alias)
+	err := r.client.StoreAccounts()
+	if err != nil {
+		log.Error("failed to create account: %v", err)
+		return
+	}
+
+	fmt.Printf("%s Created account alias: `%s`, address: %s \n", printPrefix, ac.Name, ac.Address().String())
+	r.client.SetCurrentAccount(ac)
+}
+
+// print account info from global state
+func (r *repl) printAccountInfo() {
+	acc := r.client.CurrentAccount()
+	if acc == nil {
+		r.chooseAccount()
+		acc = r.client.CurrentAccount()
+	}
+
+	address := gosmtypes.BytesToAddress(acc.PubKey)
+
+	state, err := r.client.AccountState(address)
+	if err != nil {
+		log.Error("failed to get account info: %v", err)
+		return
+	}
+
+	fmt.Println(printPrefix, "Local alias:", acc.Name)
+	fmt.Println(printPrefix, "Address:", address.String())
+	fmt.Println(printPrefix, "Balance:", state.StateCurrent.Balance.Value, coinUnitName)
+	fmt.Println(printPrefix, "Nonce:", state.StateCurrent.Counter)
+	fmt.Println(printPrefix, "Projected Balance:", state.StateProjected.Balance.Value, coinUnitName)
```

Comment:
Might be nice to add a line to output here that says something like "Projected account state includes all pending transactions that haven't been added to the mesh yet."

Reply:
done